massgen/frontend/coordination_ui.py

```
@@ -812,6 +812,43 @@ def prepare_for_restart(self, orchestrator, attempt: int, max_attempts: int) ->
+        display = self.display
+        if display is None or self.config.get("automation_mode", False):
+            return
+        if not hasattr(display, "show_tool_approval_modal"):
+            return
+        try:
+            from massgen.permissions.approval_provider import (
+                CallbackApprovalProvider,
+                PolicyApprovalProvider,
+            )
+            from massgen.permissions.models import ApprovalDecision
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)